lib/realtime/tenants/authorization/policies/presence_policies.ex

```diff
@@ -25,76 +24,66 @@ defmodule Realtime.Tenants.Authorization.Policies.PresencePolicies do
           write: boolean()
         }
   @impl true
-  def check_read_policies(_conn, policies, %Authorization{channel: nil}) do
+  def check_read_policies(_conn, policies, %Authorization{channel_name: nil}) do
     {:ok, Policies.update_policies(policies, :presence, :read, false)}
   end
 
-  def check_read_policies(conn, %Policies{} = policies, %Authorization{
-        channel: %Channel{id: channel_id}
-      }) do
-    Postgrex.transaction(conn, fn transaction_conn ->
-      query = from(b in Presence, where: b.channel_id == ^channel_id)
+  def check_read_policies(conn, %Policies{} = policies, %Authorization{channel_name: channel_name}) do
+    query =
+      from(m in Message,
+        where: m.channel_name == ^channel_name,
+        where: m.feature == :presence,
+        limit: 1
```

Comment:
do we really need a limit?

Reply:
yes, this table might have a lot of entries but we just need to fetch that the user can read the first one and if so RLS has worked and the user has the correct permissions